File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int XOR_all=0;
        for (int i=0;i<=nums.size();i++){
            XOR_all^=i;
        }
        for (int num :nums){
            XOR_all^=num;
        }
        return XOR_all;
    }
};

File: 0268-missing-number/0268-missing-number.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # n=len(nums)
        # xor_all=0
        # for i in range(n+1):
        #     xor_all^=i
        # for num in nums:
        #     xor_all^=num
 #
        # return xor_all
        n=len(nums)       
        nums_sum=sum(nums)
        original_sum=n*(n+1)//2
       
        return original_sum-nums_sum  